### ISSUE TITLE
Usage with AWS Fargate

### ISSUE THREAD

**Issue by andrewhamon** [lambci/ecs#24]
[AWS Fargate](https://aws.amazon.com/blogs/aws/aws-fargate/) was just announced, which gives per second billing on custom containers! It claims to be compatible with the existing ECS api, so it would probably be a trivial change to get it working (I say this having never set up LambCI with an ECS cluster, only Lambda, so forgive me if I am wrong).

The prospect of LambCI with none of the limitations of the Lambda environment, and still with per-second billing, sounds like a killer feature to me 😄

**Comment by mhart** [comment]
Yeah, absolutely agree – once the dust settles, will look more into this 👍

**Comment by jtyers** [comment]
ping

Would be great to see support for this. I did some initial work on this to see how far I'd get. Fargate doesn't support mounting `docker.sock` inside a container, so this would need re-engineering. As I see it, one way round it would be:

* LambCI lambda calls ECS task definition 1
* Task def 1 builds `Dockerfile.test` using a rootless builder (Buildkit/buildah?)
* Once built, create a second task definition based of the resulting new image
* Run this new task definition
* a final Lambda picks up the outcome of the task definition, and sends it to S3 or wherever

This is quite complex compared to straight Lambda sadly (the only reason I need ECS is disk space)
